### PR TITLE
tf records generation from labels.npz

### PR DESCRIPTION
**Pull request by Geoyi**
showcase using label maker with Tensorflow object detection particularly.

@drewbo, can you review this code and document? I wanna make sure this code is pythonic enough, and if you could test it out that would be great too.

### CODE REVIEW

**Comment by drewbo** [comment]
@Geoyi reading through and trying the demo now. Is there a way to write `tf_records_generation.py` so that it doesn't require `pandas`? It isn't required by the base library so I'd like to avoid it in the example (if possible)

**Comment by Geoyi** [comment]
theoretically yes @drewbo, but I only worked with dataframe/csv so far.

File: label_maker/label.py [hunk 1]
```diff
@@ -274,8 +274,8 @@ def _tile_results_summary(ml_type, classes):
     if ml_type == 'object-detection':
         # for each class, show number of features and number of tiles
         for i, cl in enumerate(classes):
-            cl_features = len([bb for l in labels for bb in l if bb[4] == i])
-            cl_tiles = len([l for l in labels if len(list(filter(_bbox_class(i), l)))]) # pylint: disable=cell-var-from-loop
+            cl_features = len([bb for l in labels for bb in l if bb[4] == i + 1])
+            cl_tiles = len([l for l in labels if len(list(filter(_bbox_class(i+1), l)))]) # pylint: disable=cell-var-from-loop
```
Comment (drewbo) [inline label_maker/label.py]: @Geoyi can you remove this commit? This change was already made on master so it's causing the merge conflict

File: examples/walkthrough-tensorflow-object-detection.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
+# Example Use: A building object detection with TensorFlow API
+
+You have to walk through all these steps to be able to train a TensorFlow Object Detection and have the following results:
```
Comment (drewbo) [inline examples/walkthrough-tensorflow-object-detection.md]: @Geoyi Can you add a short overview of what TensorFlow Object Detection API is and why we are using it?

File: examples/utils/tf_records_generation.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+    for tile in tiles:
+        bboxes = labels[tile].tolist()
+        width = 256
+        height = 256
+        if len(bboxes) != 0:
+            for bbox in bboxes:
+                if bbox[4] == 1:
+                    cl_str = "building"
+                    if bbox[0] < 0:
```
Comment (drewbo) [inline examples/utils/tf_records_generation.py]: @Geoyi can you abstract this part out into a "clamping function" like I showed earlier?

Comment (Geoyi) [reply examples/utils/tf_records_generation.py]: updated, thanks for helping out, @drewbo

File: examples/walkthrough-tensorflow-object-detection.md (hunk 2 above)
Comment (Geoyi) [reply examples/walkthrough-tensorflow-object-detection.md]: an overview has added to the head of the walkthrough too.

File: label_maker/label.py (hunk 1 above)
Comment (Geoyi) [reply label_maker/label.py]: I thought I have resolved it, will doable check.

**Pull request merged**